hypothesis-python/tests/conjecture/test_inquisitor.py

```diff
@@ -28,11 +28,14 @@ def _new():
     return _inner
 
 
+# this should have a marked as freely varying, but false negatives in our
+# inquisitor code skip over it sometimes, depending on the seen_passed_buffers.
+# yet another thing that should be improved by moving to the ir.
 @fails_with_output(
     """
 Falsifying example: test_inquisitor_comments_basic_fail_if_either(
     # The test always failed when commented parts were varied together.
-    a=False,  # or any other generated value
+    a=False,
     b=True,
     c=[],  # or any other generated value
     d=True,
```

Comment:
Yeah, you can have both false alarms and missed alarms, in both cases because the sizes can change in such a way that the suffix is identical but interpreted differently.  This _doesn't_ affect the prefix though; if two test cases have an identical prefix until some draw in one, then the other must have an equivalent draw starting at the same index.

The trouble is that to fix this requires checking against the end index in the _saved_ test case, which means we have to track the `ConjectureResult` objects rather than `bytes`, and that's more expensive.  Or, actually, maybe we can get all of this from the `DataTree`?  Hmm...

Reply:
good point about the prefix.

I think comparing datatrees would eliminate fail-to-report here. Maybe something that doesn't need to wait for the ir? (though I'm not volunteering to work on this in the immediate future 😄)

Reply:
Can you write up an issue, and then we'll ignore until it's convenient to work on?

Reply:
will do 👍